tiledb/sm/consolidator/fragment_consolidator.cc

```diff
@@ -362,7 +362,7 @@ Status FragmentConsolidator::consolidate_fragments(
   NDRange union_non_empty_domains;
   std::unordered_set<std::string> to_consolidate_set;
   for (auto& uri : fragment_uris) {
-    to_consolidate_set.emplace(uri);
+    to_consolidate_set.emplace(URI(uri).last_path_part());
```

Comment:
We should be more rigorous here and validate that the removed part is either the array URI for old fragment style or array_uri + "__fragments". Let's add coverage for all.

Reply:
Should relative fragment URIs not be allowed? I'm just thinking for that case neither of these checks would pass. The removed part of the URI would be the CWD if we do something like `URI("__1719935189448_1719935189448_2eebea0af3ad535fe64ab72eee433790_22").last_path_part()` for example.

Reply:
The old way of using the API should still work.